FWCore/Services/scripts/edmTracerCompactLogViewer.py

```diff
@@ -203,24 +238,132 @@ def textPostfix(self):
     def text(self, context):
         return f'{self.textPrefix()} {self.textSpecial()}: {self.textPostfix()}'
 
+def findMatchingTransition(sync, containers):
+    for i in range(len(containers)):
+        if containers[i][-1]["sync"] == sync:
+            return i
+    #need more exhausting search
+    for i in range(len(containers)):
+        for t in containers[i]:
+            if t["sync"] == sync:
+                return i
+
+    print("find failed",sync, containers)
+    return None
+
+def popQueuedTransitions(sync, container):
+    results = []
+    for i in range(len(container)):
+        if sync == container[i]["sync"]:
+            results.append(container[i])
+            results.append(container[i+1])
+            del container[i]
+            del container[i]
+            break
+    return results
+        
+            
 class PreFrameworkTransitionParser (FrameworkTransitionParser):
     def __init__(self, payload):
         super().__init__(payload)
     def textSpecial(self):
         return "starting"
+    def jsonInfo(self, counter, data):
+        if transitionIsGlobal(self.transition):
+            index = 0
+            if self.transition == Phase.startTracing:
+                data["globals"][0].append(jsonTransition(type=self.transition, id=index, sync=list(self.sync),start=0, finish=self.time ))
+                return
+            elif self.transition == Phase.esSync:
+                if self.sync[1] == 4294967295:
```

Comment:
What is this magic value? Could make it a named constant or add a comment?

Reply:
The number is the largest allowed number for a LuminosityBlock (it is defined in C++). I've now used a named constant.